### PR TITLE
Webc 507 token endpoint support auth code

### PR DESCRIPTION
**Pull request by augsteyer**
Please pull the changes, see if your old resource_auth_code table gets migrated into the new one successfully.

### CODE REVIEW

File: src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Checkout.php [hunk 1]
```diff
@@ -78,20 +76,26 @@ public function logoutCustomer()
     }
 
     /**
-     * @param int        $resourceId
-     * @param int | null $customerId
+     * @param int          $resourceId
+     * @param int | string $storeId
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Checkout.php]: generateToken is only called in Shopgate_Cloudapi_Model_Api2_Carts_Url_Rest::_createUrl, whereas the $storeId will be returned by Mage_Api2_Model_Resource::_getStore. This method is returning an instance of Mage_Core_Model_Store. Could you please double check?

File: src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Checkout.php [hunk 2]
```diff
@@ -78,20 +76,26 @@ public function logoutCustomer()
...
-    public function generateAuthToken($resourceId, $customerId)
+    public function generateAuthToken($resourceId, $storeId, $customerId)
     {
         /** @var Magento_Db_Adapter_Pdo_Mysql $writeConnection */
         $writeConnection = Mage::getSingleton('core/resource')->getConnection('core_write');
         /** @var Shopgate_Cloudapi_Model_OAuth2_Db_Pdo $storage */
         $storage = Mage::getModel('shopgate_cloudapi/oAuth2_db_pdo', array($writeConnection->getConnection()));
-        $token   = $storage->createAuthItemByType(
+        $storage->setStore(Mage::app()->getStore($storeId));
+
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Helper/Frontend/Checkout.php]: If my assumption from the 1st comment is correct, $storeId should already be the store instance, not need to fetch it again using getStore().

File: src/app/code/community/Shopgate/Cloudapi/Model/Api2/Carts/Url/Rest.php [hunk 3]
```diff
@@ -37,15 +38,16 @@ protected function _create(array $filteredData)
...
      *
      * @return array
+     * @throws Mage_Core_Exception
      */
     protected function _createUrl($quote)
     {
         /** @var Shopgate_Cloudapi_Helper_Frontend_Checkout $helper */
         $helper = Mage::helper('shopgate_cloudapi/frontend_checkout');
-        $token  = $helper->generateAuthToken($quote->getId(), $quote->getCustomerId() ? : null);
+        $token  = $helper->generateAuthToken($quote->getId(), $this->_getStore(), $quote->getCustomerId());
```
Comment (andre-kraus) [inline src/app/code/community/Shopgate/Cloudapi/Model/Api2/Carts/Url/Rest.php]: See comment above about return values of $this->_getStore()

Comment (augsteyer) [reply src/app/code/community/Shopgate/Cloudapi/Model/Api2/Carts/Url/Rest.php]: yes, I had a brain fart and thought I was in the controller operating with ```$request->getParam('store')``` call /facepalm

**Pull request merged**